app/src/features/apiClient/screens/apiClient/components/views/graphql/components/SchemaBuilder/SchemaBuilder.tsx

```diff
@@ -24,6 +24,30 @@ export const SchemaBuilder: React.FC<Props> = ({ setIsSchemaBuilderOpen }) => {
 
   const { introspectAndSaveSchema } = useGraphQLIntrospection();
 
+  const hasParsedSuccessfully = useRef(false);
+
+  /*
+   * This effect is added due to Explorer component's internal query caching logic.
+   * The graphiql-explorer package uses module-level memoization that caches parsed queries
+   * across all component instances. When multiple SchemaBuilder components are rendered
+   * (e.g., in different tabs), they share the same parsed query cache, causing field
+   * selection state to be shared between instances.
+   *
+   * By only passing the query to Explorer after it has been successfully parsed once,
+   * we prevent the Explorer from caching invalid queries and reduce the likelihood of
+   * shared state issues between different instances.
+   */
```

Comment:
Can we also mention the issue that cached query is used whenever the query is errored on parsing?